### PR TITLE
SCP-1825 - Split Marlowe into "Marlowe Core" and "Extended Marlowe" in Playground

### PR DESCRIPTION
**Pull request by palas**
This is a refactoring that aims to split Marlowe in two:
* "Marlowe Core" is the original `Semantics.purs`, and is used for static analysis and simulation
* and `Extended.purs` (Extended Marlowe) is, so far, essentially a copy of some key types of `Semantics.purs` that will eventually be modified to include template constructors. I have left out base constructs like `Party` and `ChoiceId`. I think it could be a good idea to eventually get rid of `Party` from "Extended Marlowe", in fact.

`Party` is a weird case, because it is not in the "Extended Marlowe" module but, because it has holes, I had to create two parsers for it, one that accepts holes and one that doesn't. Because I can only generate "Extended Marlowe" from the parser with holes, and Party is also in things like Inputs, which are not in "Extended Marlowe". Also `AccountId` because it is a type synonym of `Party`

"Extended Marlowe" is used by linting, and the hole system, and as a target for Haskell, JavaScript, and Blockly.

There are at least a few things that probably just work for now because I haven't changed "Extended Marlowe" to be different from "Marlowe Core":
* Starting the simulation, will need an interface for conversion (also the wallet, even though it is hidden right now)
* Linting, highlighting of reachability analysis
* Static Analysis, will also need an interface for conversion
* Serialisation-Deserialisation tests
* Generators

Deployed to https://pablo.marlowe.iohkdev.io/

Pre-submit checklist:
- Branch
    - [x] Commit sequence broadly makes sense
    - [x] Key commits have useful messages
    - [x] Relevant tickets are mentioned in commit messages
- PR
    - [x] Self-reviewed the diff
    - [x] Useful pull request description
    - [x] Reviewer requested
- If you updated any cabal files or added Haskell packages:
    - [x] `nix-shell shell.nix --run updateMaterialized` to update the materialized Nix files
    - [x] Update `hie-*.yaml` files if needed
- If you changed any Haskell files:
    - [x] `nix-shell shell.nix --run fix-stylish-haskell` to fix any formatting issues
- If you changed any Purescript files:
    - [x] `nix-shell shell.nix --run fix-purty` to fix any formatting issues

Pre-merge checklist:
- [x] Someone approved it
- [x] Commits have useful messages
- [x] Review clarifications made it into the code
- [x] History is moderately tidy; or going to squash-merge

### CODE REVIEW

**Review by shmish111 — CHANGES_REQUESTED** [review]
apart from my small comments, I suggest we use `Marlowe.Extended` rather than `Marlowe.ExtendedMarlowe`

File: marlowe-playground-client/src/Marlowe/ExtendedMarlowe.purs [hunk 1]
```diff
@@ -0,0 +1,519 @@
...
+  show v = genericShow v
+
+instance prettyContract :: Pretty Contract where
+  pretty v = genericPretty v
+
+instance hasArgsContract :: Args Contract where
+  hasArgs a = genericHasArgs a
+  hasNestedArgs a = genericHasNestedArgs a
+
+convertPayeeIfNoExtensions :: Payee -> Maybe S.Payee
```
[the remark this answers is not in the document]
Comment (palas) [reply marlowe-playground-client/src/Marlowe/ExtendedMarlowe.purs:445]: I am not sure what you mean. What would the class be?

[the remark this answers is not in the document]
Comment (palas) [reply marlowe-playground-client/src/Marlowe/ExtendedMarlowe.purs:445]: Do you mean a nested function?

[the remark this answers is not in the document]
Comment (shmish111) [reply marlowe-playground-client/src/Marlowe/ExtendedMarlowe.purs]: e.g.
```purescript
class ToCore a b where
  toCore :: a -> Maybe b
```

It just makes all the function names shorter, i.e. `convertValueIfNoExtensions` becomes `toCore`

**Review by shmish111 — APPROVED** [review]

**Pull request merged**